### PR TITLE
refactor(migrations): validator

### PR DESCRIPTION
**Pull request by pret-a-porter**
## Changes:

1. Introduce custom migration validator to test only one specific migration. It increases unit test speed and reduces code duplication
2. Add few more migrations from previously huge PR https://github.com/renovatebot/renovate/pull/12957
3. Replace `delete` method by `deprecated` field for more declarative way
4. Add debug task "Jest Current Folder" for vscode, which runs all tests inside selected directory

## Context:

1. https://github.com/renovatebot/renovate/issues/11459
2. Part of https://github.com/renovatebot/renovate/pull/12957 

## Documentation

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -137,6 +137,12 @@ module.exports = {
         '@typescript-eslint/explicit-function-return-type': 0,
         '@typescript-eslint/unbound-method': 0,
 
+        'jest/expect-expect': [
+          'error',
+          {
+            assertFunctionNames: ['expect', 'validate'],
```
Comment (viceice) [inline .eslintrc.js:143]: Why this?

File: .vscode/launch.json [hunk 2]
```diff
@@ -60,6 +60,29 @@
...
+      "args": [
+        "--runInBand",
+        "--collectCoverage=false",
+        "--testTimeout=100000000",
+        "--roots=${workspaceFolder}/${relativeFileDirname}"
+      ],
+      "env": {
+        "NODE_ENV": "test",
+        "LOG_LEVEL": "debug"
+      },
```
Comment (viceice) [inline .vscode/launch.json:77]:
```suggestion
```
Remove

**Review by viceice — COMMENTED** [review]
Please split refactoring from new migration to separate PR.

File: .eslintrc.js (hunk 1 above)
Comment (pret-a-porter) [reply .eslintrc.js:143]: Otherwise eslint trows an error, it expects `expect` keywork inside each unit test by default

**Comment by pret-a-porter** [comment]
> Please split refactoring from new migration to separate PR.

I have removed commits with migrations, but unfortunately by force push

File: .eslintrc.js (hunk 1 above)
Comment (viceice) [reply .eslintrc.js:143]: Yes, that should be enforced, please only mark required tests and don't disable this globally.

Comment (pret-a-porter) [reply .eslintrc.js:143]: Yeah, I did not disable anything. I only extended allowed keywords, now it is no only `expect`, but `validate` as well

Comment (rarkins) [reply .eslintrc.js:148]: So with this change, either `expect` or `validate` are allowed *globally*? @viceice is that a concern if so? @pret-a-porter alternatively is it too hard to set this value just where applicable?

Comment (pret-a-porter) [reply .eslintrc.js:148]: Yes, now expect or validate are allowed globally. I am not sure, that possible to apply this eslint rule only for specific folder.

Comment (rarkins) [reply .eslintrc.js:148]: @viceice does this sound ok with you?

Comment (pret-a-porter) [reply .eslintrc.js:148]: Done

File: lib/config/migrations/base/remove-property-migration.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ export class RemovePropertyMigration extends AbstractMigration {
     this.propertyName = propertyName;
   }
 
-  override run(): void {
+  run(): void {
```
Comment (viceice) [inline lib/config/migrations/base/remove-property-migration.ts:16]: Why override is no longer required?

File: lib/config/migrations/custom/binary-source-migration.spec.ts [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
-      binarySource: 'global',
-    });
+    expect(BinarySourceMigration).toMigrate(
+      {
+        binarySource: 'auto',
+      },
+      {
+        binarySource: 'global',
+      }
+    );
```
Comment (viceice) [inline lib/config/migrations/custom/binary-source-migration.spec.ts:12]: That looks good 👍

File: lib/config/migrations/base/remove-property-migration.ts (hunk 3 above)
Comment (pret-a-porter) [reply lib/config/migrations/base/remove-property-migration.ts:16]: I decided that is not correct to use `override` keyword for methods, which is marked as `abstract` in parent class

Comment (viceice) [reply lib/config/migrations/base/remove-property-migration.ts:16]: Can you check if we set the `noImplicitOverride` typescript option? It should be enabled and force the override usage.

**Review by viceice — COMMENTED** [review]
See comments

File: lib/config/migrations/base/remove-property-migration.ts (hunk 3 above)
Comment (pret-a-porter) [reply lib/config/migrations/base/remove-property-migration.ts:16]: Yeah, it is already enabled 5 month ago

Comment (viceice) [reply lib/config/migrations/base/remove-property-migration.ts:16]: I think we should leave the override, so we can't get into trouble in future

Comment (pret-a-porter) [reply lib/config/migrations/base/remove-property-migration.ts]: Returned

**Review by viceice — APPROVED** [review]
🥳

**Pull request merged**